examples/federated-learning/tff/nvflare/README.md

```diff
@@ -0,0 +1,212 @@
+
+![alt_text](../../../../assets/nvflare.svg "Infrastructure overview")
+
+As shown in the preceding diagram, the blueprint helps you to create and configure the following infrastructure components:
+
+- A persistent volume to store the Nvflare workspace
+- Two pods that are the clients that will be connected to the server in the nvidia-client1 and nvidia-client2 namespaces respectively
+- One pod that is the the server that will aggregate all the results from the computation in the nvflare-infra namespace
+
+## 1. Create custom image
+
+For this demo to work, you have to create a custom image with TensorFlow and Nvflare installed. To create the image and push it on Artifact Registry, go to the `docker-image` and build the image:
+
+```bash
+cd docker-image
+export REGION=$(gcloud config get compute/region)
+export PROJET_ID=$(gcloud config get core/project)
+export REPOSITORY=my-repository
+gcloud builds submit --tag ${REGION}-docker.pkg.dev/${PROJECT_ID}/${REPOSITORY}/nvflare-tensorflow:1.0.0
+```
+
+Once the image is built, modify the `kustomization.yaml` file to add the new name and new tag of the image.
+
+## 2. Create persistent volume for Nvflare workspace
+
+All the models generated will be stored in a Cloud storage bucket mounted by each pod. First, create a bucket:
+
+```bash
+gcloud storage buckets create gs://nvflare-storage
+```
+
+Add the following permissions to the Kubernetes Service Account:
+
+```bash
+gcloud storage buckets add-iam-policy-binding gs://nvflare-storage \
+    --member "principal://iam.googleapis.com/projects/391818703174/locations/global/workloadIdentityPools/lgu-demos.svc.id.goog/subject/ns/default/sa/default" \
+    --role "roles/storage.objectUser"
+```
+
+## 3. Create Nvflare folder structure
+
+Now that the persistent volume is created, you can start creating the folder structure and upload it to GCS. First install nvflare on your workstation:
+
+```bash
+python3 -m pip install -r requirements.txt
+```
+
+Then, run the provisionning tool:
+
+```bash
+cd ${HOME}
+nvflare provision
+```
+
+You have to choose the non-ha (non high-availability) choice, as ha (high-availability) is not yet supported on Kubernetes. It will create a `project.yml` file that you can customize for your needs. Then, re-run the provisionning tool:
+
+```bash
+cd ${HOME}
+nvflare provision
+cd ${HOME}/workspace/example_project/prod_00/
+```
+
+The different folders generated represent the infrastructure you will deploy on the reference architecture:
+
+- `server1` is the server that will aggregate all the results from the computation
+- `site-1` and `site-2` are the clients that will be connected to the server
+- `admin@nvidia.com` is the administration client to start and list jobs
+
+## 4. Clone the repository
+
+You will need to clone the Nvflare repository that contains the job you will run on the reference architecture:
+
+```bash
+cd ${HOME}
+git clone https://github.com/NVIDIA/NVFlare
```

Comment:
If you go the `git clone` approach, you should checkout a specific tag so this procedure is more reproducible.

Reply:
Is it better ?